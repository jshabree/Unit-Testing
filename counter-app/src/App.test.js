// test that app renders without crashing 

import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// dummy test

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2+2).toBe(4);
  });
});

// it and test are aliases

test('knows that 2 and 2 make 4', () => {
  expect(2 + 2).toBe(4);  // matcher function
});


// testing app's initial state using shallow rendering

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
    });
});

// testing user's interaction to see if count is indeed incremented ( in this case, by 1)

describe('App', () => {
  it('increments count by 1 when button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrementButton = wrapper.find('button.increment');
    incrementButton.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
  });
});