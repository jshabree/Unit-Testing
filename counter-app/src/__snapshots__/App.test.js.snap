// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment value of count
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement value of count
  </button>
</div>
`;

exports[`matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment value of count
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement value of count
  </button>
</div>
`;
